win32ss/user/user32/windows/cursoricon.c

```diff
@@ -1418,15 +1632,35 @@ CURSORICON_LoadFromFileW(
         cursorData.xHotspot = entry->xHotspot;
         cursorData.yHotspot = entry->yHotspot;
     }
-    cursorData.rt = (USHORT)((ULONG_PTR)(bIcon ? RT_ICON : RT_CURSOR));
+    cursorData.rt = LOWORD(bIcon ? RT_ICON : RT_CURSOR);
 
-    /* Do the dance */
-    if(!CURSORICON_GetCursorDataFromBMI(&cursorData, (BITMAPINFO*)(&bits[entry->dwDIBOffset])))
+    /* Try to load BMP icon */
+    if (!CURSORICON_GetCursorDataFromBMI(&cursorData, (PBITMAPINFO)(&bits[entry->dwDIBOffset])))
+    {
+        /* Convert PNG raw data to BMP icon if the icon was PNG icon */
+        PBYTE pngBits = &bits[entry->dwDIBOffset];
+        pbBmpIcon = CURSORICON_ConvertPngToBmpIcon(pngBits, filesize, &BmpIconSize);
+        if (!pbBmpIcon)
+            goto end; /* Not PNG icon or failed */
+
+        /* Find icon entry from BMP icon */
+        dir = (CURSORICONFILEDIR *)pbBmpIcon;
+        entry = &dir->idEntries[0]; /* Only one entry */
+
+        /* A bit of preparation */
+        ZeroMemory(&cursorData, sizeof(cursorData));
+        cursorData.rt = LOWORD(bIcon ? RT_ICON : RT_CURSOR);
```

Comment:
I don't know whether it's intended here, but by ZeroMemory-ing cursorData, you erase what has been set for the xHotspot and yHotspot at line https://github.com/reactos/reactos/pull/7709/files#diff-0bde9244bc3fdf95faed2fc129956a4ce8e298dd37f39d9ad67030e7988d6bf6R1632 .
(And you are forced to redo the cursorData.rt = .... with the same code.)